yellowbrick/features/manifold.py

```diff
@@ -299,179 +332,103 @@ def fit_transform(self, X, y=None):
         """
         Returns the transformed data points from the manifold embedding.
 
         Parameters
         ----------
         X : array-like of shape (n, m)
             A matrix or data frame with n instances and m features
-
+            
+        y : array-like of shape (n,), optional
+            The target, used to specify the colors of the points.
+            
         Returns
         -------
         Xprime : array-like of shape (n, 2)
             Returns the 2-dimensional embedding of the instances.
+            
+        Note
+        ----
+        This method does not work with MDS, TSNE and SpectralEmbedding because 
+        it is yet to be implemented in sklearn.
         """
         try:
-            return self.manifold.transform(X)
+            Xp = self.manifold.transform(X)
         except AttributeError as e:
             raise AttributeError(str(e) + " try using fit_transform instead.")
 
-    def draw(self, X, y=None):
-        """
-        Draws the points described by X and colored by the points in y. Can be
-        called multiple times before finalize to add more scatter plots to the
-        axes, however ``fit()`` must be called before use.
-
-        Parameters
-        ----------
-        X : array-like of shape (n, 2)
-            The matrix produced by the ``transform()`` method.
-
-        y : array-like of shape (n,), optional
-            The target, used to specify the colors of the points.
-
-        Returns
-        -------
-        self.ax : matplotlib Axes object
-            Returns the axes that the scatter plot was drawn on.
-        """
-        scatter_kwargs = {"alpha": self.alpha}
-
-        # Determine the colors
-        if self._target_color_type == SINGLE:
-            scatter_kwargs["c"] = "b"
-
-        elif self._target_color_type == DISCRETE:
-            if y is None:
-                raise YellowbrickValueError("y is required for discrete target")
-
-            scatter_kwargs["c"] = [
-                self._colors[np.searchsorted(self.classes_, (yi))] for yi in y
-            ]
-
-        elif self._target_color_type == CONTINUOUS:
-            if y is None:
-                raise YellowbrickValueError("y is required for continuous target")
-
-            # TODO manually make colorbar so we can draw it in finalize
-            scatter_kwargs["c"] = y
-            scatter_kwargs["cmap"] = self.colors or palettes.DEFAULT_SEQUENCE
-
-        else:
-            # Technically this should never be raised
-            raise NotFitted("could not determine target color type")
+        self.draw(Xp, y, **kwargs)
+        return Xp
 
-        # Draw the scatter plot with the associated colors and alpha
-        self._scatter = self.ax.scatter(X[:,0], X[:,1], **scatter_kwargs)
-        return self.ax
+    def draw(self, Xp, y=None):
+        # Calls draw method from super class which draws scatter plot.
+        super(Manifold, self).draw(Xp, y)
```

Comment:
This is probably not necessary but I like that you've added it to show explicitly that we're using the superclass draw method - very classy! (pun intended)

Reply:
Hehe! Thanks!